packages/odyssey-react-mui/src/OdysseyProvider.tsx

```diff
@@ -51,32 +62,73 @@ const OdysseyProvider = <SupportedLanguages extends string>({
   translationOverrides,
-}: OdysseyProviderProps<SupportedLanguages>) => (
-  <OdysseyCacheProvider
-    emotionRoot={emotionRoot}
-    emotionRootElement={emotionRootElement}
-    hasShadowDom={Boolean(shadowRootElement || shadowDomElement)}
-    nonce={nonce}
-    stylisPlugins={stylisPlugins}
-  >
-    <OdysseyThemeProvider
-      designTokensOverride={designTokensOverride}
-      shadowRootElement={shadowRootElement || shadowDomElement}
-      themeOverride={themeOverride}
+}: OdysseyProviderProps<SupportedLanguages>) => {
+  const { contrastMode } = useContrastContext();
+
+  const odysseyTokens = useMemo(
+    () => ({ ...Tokens, ...designTokensOverride }),
+    [designTokensOverride],
+  );
+
+  const odysseyTheme = useMemo(
+    () =>
+      createOdysseyMuiTheme({
+        odysseyTokens,
+        shadowRootElement: shadowRootElement || shadowDomElement,
+      }),
+    [odysseyTokens, shadowDomElement, shadowRootElement],
+  );
+
+  const odysseyThemeWithBackground = useMemo(
+    () =>
+      createTheme({
+        ...odysseyTheme,
+        odysseyContrastMode: contrastMode,
+      }),
+    [odysseyTheme, contrastMode],
+  );
+
+  const customOdysseyTheme = useMemo(
+    () =>
+      themeOverride
+        ? createTheme(deepmerge(odysseyThemeWithBackground, themeOverride))
+        : odysseyThemeWithBackground,
+    [odysseyThemeWithBackground, themeOverride],
+  );
+
+  return (
+    <OdysseyCacheProvider
+      emotionRoot={emotionRoot}
+      emotionRootElement={emotionRootElement}
+      hasShadowDom={Boolean(shadowRootElement || shadowDomElement)}
+      nonce={nonce}
+      stylisPlugins={stylisPlugins}
     >
-      {/* This component creates a div; for flexibility of layout of children, make it inherit its parent's height */}
-      <ScopedCssBaseline sx={scopedCssBaselineStyles}>
-        <OdysseyTranslationProvider<SupportedLanguages>
-          languageCode={languageCode}
-          translationOverrides={translationOverrides}
-        >
-          {children}
-        </OdysseyTranslationProvider>
-      </ScopedCssBaseline>
-    </OdysseyThemeProvider>
-  </OdysseyCacheProvider>
+      <MuiThemeProvider theme={customOdysseyTheme}>
+        <OdysseyDesignTokensContext.Provider value={odysseyTokens}>
+          {/* This component creates a div; for flexibility of layout of children, make it inherit its parent's height */}
+          <ScopedCssBaseline sx={scopedCssBaselineStyles}>
+            <OdysseyTranslationProvider<SupportedLanguages>
+              languageCode={languageCode}
+              translationOverrides={translationOverrides}
+            >
+              {children}
+            </OdysseyTranslationProvider>
+          </ScopedCssBaseline>
+        </OdysseyDesignTokensContext.Provider>
+      </MuiThemeProvider>
+    </OdysseyCacheProvider>
+  );
+};
+
+const OdysseyProvider = <SupportedLanguages extends string>(
+  props: OdysseyProviderProps<SupportedLanguages>,
+) => (
+  <ContrastModeProvider>
```

Comment:
@benlister-okta You can't currently set an explicit contrast mode. Looking at the code it seems like something you were trying to support

Reply:
You can by passing `contrastMode="lowContrast"` (or highContrast) to the provider. The logic for it is in the ContrastProvider file: https://github.com/okta/odyssey/pull/2364/files#diff-69196edb1faa4f2d8100e2344e3ebd81a965621255c029d9380c8a5f976d2901

in this block `
  useLayoutEffect(() => {
    if (explicitContrastMode) {
      setContrastMode(explicitContrastMode);
    } else {
      const isLowContrast = parentBackgroundColor === Tokens.HueNeutral50;
      setContrastMode(isLowContrast ? "lowContrast" : "highContrast");
    }
  }, [parentBackgroundColor, explicitContrastMode]);`

Reply:
Right but, you are not passing that prop through here.

Reply:
ahh, I see what you mean now, I added it in there now too. I was testing in storybook by importing directly so I didn't notice.